internal/databases/postgres/daemon_handler.go

```diff
@@ -0,0 +1,176 @@
+
+func (h *ArchiveMessageHandler) Handle(messageBody []byte) error {
+	err := messageValidation(messageBody)
+	if err != nil {
+		tracelog.ErrorLogger.Printf("Incorrect message: %v\n", err)
+		return err
+	}
+	tracelog.InfoLogger.Printf("wal file name: %s\n", string(messageBody))
+	PgDataSettingString, ok := internal.GetSetting(internal.PgDataSetting)
+	if !ok {
+		tracelog.ErrorLogger.Print("\nPGDATA is not set in the conf.\n")
+		return fmt.Errorf("PGDATA is not set in the conf")
+	}
+	pathToWal := path.Join(PgDataSettingString, "pg_wal")
+	fullPath := path.Join(pathToWal, string(messageBody))
+	tracelog.InfoLogger.Printf("starting wal-push for %s\n", fullPath)
+	err = HandleWALPush(h.uploader, fullPath)
+	if err != nil {
+		tracelog.ErrorLogger.Printf("Failed to archive file: %s, err: %v \n", string(messageBody), err)
+		return err
+	}
+	_, err = h.fd.Write([]byte{'O'})
+	if err != nil {
+		tracelog.ErrorLogger.Printf("failed to write in socket: %v\n", err)
+		return err
+	}
+	return nil
+}
+
+func messageValidation(messageBody []byte) error {
+	if len(messageBody) < 24 {
+		if len(messageBody) > 0 {
+			tracelog.ErrorLogger.Println("incorrect message accepted")
+			return fmt.Errorf("incorrect message accepted: %s", string(messageBody))
+		}
+		return fmt.Errorf("empty message accepted")
+	}
+	return nil
+}
+
+func GetMessageHandler(messageType byte, c net.Conn, uploader *WalUploader) SocketMessageHandler {
+	switch messageType {
+	case 'C':
+		return &CheckMessageHandler{CheckType, c, uploader}
+	case 'F':
+		return &ArchiveMessageHandler{FileNameType, c, uploader}
+	default:
+		return nil
+	}
+}
+
+type SocketMessageReader struct {
+	c net.Conn
+}
+
+func GetMessageReader(c net.Conn) *SocketMessageReader {
+	return &SocketMessageReader{c}
+}
+
+// Next method reads messages sequentially from the Reader
+func (r SocketMessageReader) Next() (messageType byte, messageBody []byte, err error) {
+	messageParameters := make([]byte, 3)
+	_, err = io.ReadFull(r.c, messageParameters)
+	if err != nil {
+		tracelog.ErrorLogger.Printf("Failed to read from socket, err: %v \n", err)
+		return 'E', nil, err
+	}
+	messageType = messageParameters[0]
+	var messageLength uint16
+	l := bytes.NewReader(messageParameters[1:3])
+	err = binary.Read(l, binary.BigEndian, &messageLength)
+	if err != nil {
+		tracelog.ErrorLogger.Printf("Failed to read message length, err: %v \n", err)
+		return 'E', nil, err
+	}
+	messageBody = make([]byte, messageLength-3)
+	_, err = io.ReadFull(r.c, messageBody)
+	if err != nil {
+		tracelog.ErrorLogger.Printf("Failed to read from socket, err: %v \n", err)
+		return 'E', nil, err
+	}
+	return messageType, messageBody, err
+}
+
+// HandleDaemon is invoked to perform daemon mode
+func HandleDaemon(uploader *WalUploader, pathToSocket string) {
+	_ = os.Remove(pathToSocket)
+	l, err := net.Listen("unix", pathToSocket)
+	if err != nil {
+		tracelog.ErrorLogger.Fatal("Error on listening socket:", err)
+	}
+	for {
+		fd, err := l.Accept()
```

Comment:
What if we would want to terminate the running daemon? 

I suggest to add the signal listener and listend for `SIGTERM` to perform a graceful shutdown. But looks like it is a feature for the next PR, not for the MVP.